primitives/types/account_id.go

```diff
@@ -0,0 +1,58 @@
+package types
+
+import (
+	"bytes"
+	"errors"
+	"reflect"
+
+	sc "github.com/LimeChain/goscale"
+)
+
+var (
+	errorPubKeyNotSupported = errors.New("public key type not supported")
+)
+
+type Signer interface {
+	Ed25519Signer | Sr25519Signer | EcdsaSigner
+}
+
+// AccountId It's an account ID (pubkey).
+type AccountId struct {
+	Ed25519Signer
+	Sr25519Signer
+	EcdsaSigner
+}
+
+func (a AccountId) Encode(buffer *bytes.Buffer) {
+	a.Ed25519Signer.Encode(buffer)
```

Comment:
this works, since all three fields are of type `FixedSequence[U8]`, which does not result in additional bytes  when encoding an empty field

but maybe we need to make the fields private and provide an initializer, so there is not way to set two different fields at same time, which will break the encoding 

or take a different approach and have only one field of specific type `Ed25519Signer/Sr25519Signer/EcdsaSigner`

it might require additional interface (not the `Signer` constraint) implemented by all tree signature types

```go

type SignerAddress interface {
	sc.Encodable

	SignatureType() uint
}

const (
	Ed25519 = 0
	Sr25519 = 1
	Ecdsa   = 2
)

func (s Ed25519Signer) SignatureType() uint {
	return Ed25519
}

func (s Sr25519Signer) SignatureType() uint {
	return Sr25519
}

func (s EcdsaSigner) SignatureType() uint {
	return Ecdsa
}

type AccountId[S SignerAddress] struct {
	signerAddress S
}

func New[S SignerAddress](signer S) (AccountId[S], error) {
	return AccountId[S]{signerAddress: signer}, nil
}

func (a AccountId[S]) Encode(buffer *bytes.Buffer) {
	a.signer.Encode(buffer)
}
```